plugins/extractors/bigquery/bigquery.go

```diff
@@ -311,7 +335,8 @@ func (e *Extractor) buildPreview(ctx context.Context, t *bigquery.Table) (fields
 	tempRows := []interface{}{}
 	totalRows := 0
 	ri := t.Read(ctx)
-	ri.PageInfo().MaxSize = e.getMaxPageSize()
+	// fetch only the required amount of rows
+	ri.PageInfo().MaxSize = e.config.MaxPreviewRows
```

Comment:
Not really. since we only need the `MaxPreviewRows` count, it is better to directly set that number as max page size.

Ex:
If `MaxPreviewRows` = 10, and `e.getMaxPageSize()` = 5 -> we are better off using `MaxPreviewRows`
If `MaxPreviewRows` = 20, and `e.getMaxPageSize()` = 50 -> we are better off using `MaxPreviewRows`

Reply:
If `MaxPreviewRows` is a very large number it would be better to respect the `e.getMaxPageSize()`. It would be a bit surprising that we ignore the `e.getMaxPageSize()` for this fetch operation.